### ISSUE TITLE
same config WinSW.NETCore31.x64.exe work but WinSW.NETCore31.zip  doesn't

### ISSUE THREAD

**Issue by silenceleaf** [winsw/winsw#432]
Same application config using standalone exe file everything work, but the  WinSW.NETCore31.zip  doesn't. DotNet version 3.1.2

2020-03-04 20:21:18,865 FATAL - WMI Operation failure: ServiceRequestTimeout
WMI.WmiException: ServiceRequestTimeout
   at WMI.WmiRoot.BaseHandler.CheckError(ManagementBaseObject result) in C:\projects\winsw-release\src\Core\WinSWCore\Wmi.cs:line 147
   at WMI.WmiRoot.InstanceHandler.Invoke(Object proxy, MethodInfo method, Object[] arguments) in C:\projects\winsw-release\src\Core\WinSWCore\Wmi.cs:line 236
   at WMI.Win32ServiceProxy.StartService()
   at winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor) in C:\projects\winsw-release\src\Core\ServiceWrapper\Main.cs:line 732
   at winsw.WrapperService.Main(String[] args) in C:\projects\winsw-release\src\Core\ServiceWrapper\Main.cs:line 495
WMI.WmiException: ServiceRequestTimeout
   at WMI.WmiRoot.BaseHandler.CheckError(ManagementBaseObject result) in C:\projects\winsw-release\src\Core\WinSWCore\Wmi.cs:line 147
   at WMI.WmiRoot.InstanceHandler.Invoke(Object proxy, MethodInfo method, Object[] arguments) in C:\projects\winsw-release\src\Core\WinSWCore\Wmi.cs:line 236
   at WMI.Win32ServiceProxy.StartService()
   at winsw.WrapperService.Run(String[] _args, ServiceDescriptor descriptor) in C:\projects\winsw-release\src\Core\ServiceWrapper\Main.cs:line 732
   at winsw.WrapperService.Main(String[] args) in C:\projects\winsw-release\src\Core\ServiceWrapper\Main.cs:line 495

**Comment by NextTurn** [comment]
I'm dropping the runtime-dependent zip in #421 with #411 and #416 in mind. The final plan is to get rid of WMI.

**Comment by silenceleaf** [comment]
> I'm dropping the runtime-dependent zip in #421 with #411 and #416 in mind. The final plan is to get rid of WMI.

I think it is ok to keep the runtime dependent zip release. I am curious why multiple services can not share same binary but using different config XML. Right now, I have to copy the binary everywhere.

**Comment by NextTurn** [comment]
> I am curious why multiple services can not share same binary but using different config XML. Right now, I have to copy the binary everywhere.

Specifying config files via command line is something I want to tackle next. #409 

> I think it is ok to keep the runtime dependent zip release.

The Runtime-dependent artifact can be published to Nuget instead. If you have .NET Core installed you will be able to run a command like `dotnet tool install winsw` to download it. How to do like this idea?

**Comment by silenceleaf** [comment]
> > I am curious why multiple services can not share same binary but using different config XML. Right now, I have to copy the binary everywhere.
> 
> Specifying config files via command line is something I want to tackle next. #409
> 
> > I think it is ok to keep the runtime dependent zip release.
> 
> The runtime-dependent artifact can be published to Nuget instead. If you have .NET Core installed you will be able to run a command like `dotnet tool install winsw` to download it. How to do like this idea?

Sounds good!

Looking forward to the config file path parameter!

**Comment by NextTurn** [comment]
@silenceleaf `ServiceRequestTimeout` often happens when the service crashes before starting. Since this issue was opened we've only spotted one bug (#511) about *.exe* vs *.zip* which calculated the wrong working directory for *.exe*.

Can you still reproduce this issue? If so could you check the *.wrapper.log* to see if the service crashes?

**Comment by silenceleaf** [comment]
This is reported long time ago.  Per @NextTurn  's reply.  

The final plan is to get rid of WMI.

I guess you don't need to fix it.

**Comment by NextTurn** [comment]
This might not be a WMI issue... WMI wraps native APIs and "ServiceRequestTimeout" is essentially "Error 1053: The service did not respond to the start or control request in a timely fashion." which often happens if the service hangs or crashes before responding to SCM.

Feel free to update if you still experience this issue.

**Comment by Louis-Tian** [comment]
I am experiencing the opposite. 
For release 3.0.0-alpha.6, the WinSW.NETCore.x64.exe doesn't work for me on Windows 8.1, but the WinSW.NETCore.x64.zip does. And both works just fine on a Windows 10 machine.

**Comment by NextTurn** [comment]
@Louis-Tian Are you experiencing the same issue as https://github.com/winsw/winsw/issues/432#issue-575984074 described? The error messages should be different.

**Comment by Louis-Tian** [comment]
@NextTurn Somehow I can no longer replicate the error anymore. I will try to describe what happened for whatever it's worth. 

1. I used the WinSW.NETCore.x64.exe to install the service
2. I tried to start the service via the Windows Service Manager GUI
3. I got error "Error 1053: The service did not respond to the start or control request in a timely fashion"
4. This last error message is what led me to this issue.
5. I replaced the WinSW.NETCore.x64.exe with WinSW.NETCore.x64.zip, I was able to get things working from both command line and GUI.
6. Now, if I reinstall the service with WinSW.NETCore.x64.exe, everything is still working just fine.
